### PR TITLE
Ele 1684 create an unlinked tables macro

### PR DESCRIPTION
**Pull request by ofek1weiss**

### CODE REVIEW

File: macros/commands/get_tables_without_model.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  {% endfor %}
+{% endmacro %}
+
+
+{% macro get_db_tables_in_schemas(schemas) %}
+  {% set tables = [] %}
+  {% for schema in schemas %}
+    {% set db_name, schema_name = schema.split('.') %}
+    {% set schema_relation = api.Relation.create(db_name, schema_name).without_identifier() %}
+    {% set relations = list_relations_without_caching(schema_relation) %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:17]: Prefix package

Comment (ofek1weiss) [reply macros/commands/get_tables_without_model.sql]: DONE

File: macros/commands/get_tables_without_model.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{% macro get_tables_without_models(deprecated_models_path=none) %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:1]: From this macro you could infer that it's somewhat related to sourced, snapshots, seeds, etc.
Maybe call it something in the area of `leftover` or `dangling`?
Not super critical I suppose.

File: macros/commands/get_tables_without_model.sql [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+
+{% macro get_db_tables_in_schemas(schemas) %}
+  {% set tables = [] %}
+  {% for schema in schemas %}
+    {% set db_name, schema_name = schema.split('.') %}
+    {% set schema_relation = api.Relation.create(db_name, schema_name).without_identifier() %}
+    {% set relations = dbt.list_relations_without_caching(schema_relation) %}
+    {# list_relations_without_caching can return either a list of Relation objects or an agate depending on the adapter #}
+    {% if relations.append is defined %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:19]: Why not do `relations is list`?

File: macros/commands/get_tables_without_model.sql [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      {# relations is a list of Relation objects #}
+      {% for relation in relations %}
+        {% do tables.append(schema ~ "." ~ relation.identifier) %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:22]: I usually prefer using `api.Relation.create` as it takes care of stringifying the relation in the safest way (quoting, etc.).

File: macros/commands/get_tables_without_model.sql [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  {% set db_tables = elementary.get_db_tables_in_schemas(model_schemas) %}
+  {% for db_table in db_tables %}
+    {% if db_table not in model_tables %}
+      {% do print(db_table) %}
+    {% endif %}
+  {% endfor %}
+{% endmacro %}
+
+
+{% macro get_db_tables_in_schemas(schemas) %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:12]: What do you think about calling it `get_tables_in_schemas`?
The `db` confused me a bit 😅

File: macros/commands/get_tables_without_model.sql [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+{% macro get_model_schemas_and_tables(deprecated_models_path=none) %}
+  {% set model_schemas = [] %}
+  {% set model_tables = [] %}
+
+  {% set relevant_nodes = [] %}
+  {% for model_node in graph.nodes.values() | selectattr('resource_type', '==', 'model') %}
+    {% if not deprecated_models_path %}
+      {% do relevant_nodes.append(model_node) %}
+    {% endif %}
+    {% if deprecated_models_path and not model_node.original_file_path.startswith(deprecated_models_path) %}
```
Comment (elongl) [inline macros/commands/get_tables_without_model.sql:44]:
```suggestion
    {% elif not model_node.original_file_path.startswith(deprecated_models_path) %}
```

File: macros/commands/get_tables_without_model.sql (hunk 3 above)
Comment (ofek1weiss) [reply macros/commands/get_tables_without_model.sql:19]: this is pretty hacky, but there isnt an `is list` test in jinja, only an `is iterable`, and an agate is also iterable.
i tried to find a less hacky solution but this is the best i found

File: macros/commands/get_tables_without_model.sql (hunk 2 above)
Comment (ofek1weiss) [reply macros/commands/get_tables_without_model.sql:1]: i like dangling, on it

File: macros/commands/get_dangling_tables.sql [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    {% if not deprecated_models_path %}
+      {% do relevant_nodes.append(model_node) %}
+    {% elif not model_node.original_file_path.startswith(deprecated_models_path) %}
+      {% do relevant_nodes.append(model_node) %}
+    {% endif %}
+  {% endfor %}
+  {% for model_node in relevant_nodes %}
+    {% set model_schema = api.Relation.create(model_node.database, model_node.schema).without_identifier() %}
+    {% set model_table = api.Relation.create(model_node.database, model_node.schema, model_node.name) %}
+    {% if model_schema not in model_schemas %}
```
Comment (elongl) [inline macros/commands/get_dangling_tables.sql:52]: You can use a `set()` if you want but not critical.

File: macros/commands/get_dangling_tables.sql [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  {% for model_node in graph.nodes.values() | selectattr('resource_type', '==', 'model') %}
+    {% if not deprecated_models_path %}
+      {% do relevant_nodes.append(model_node) %}
+    {% elif not model_node.original_file_path.startswith(deprecated_models_path) %}
+      {% do relevant_nodes.append(model_node) %}
+    {% endif %}
+  {% endfor %}
+  {% for model_node in relevant_nodes %}
+    {% set model_schema = api.Relation.create(model_node.database, model_node.schema).without_identifier() %}
+    {% set model_table = api.Relation.create(model_node.database, model_node.schema, model_node.name) %}
```
Comment (elongl) [inline macros/commands/get_dangling_tables.sql:51]: You need to use [`alias`](https://docs.getdbt.com/docs/build/custom-aliases), not `name`.

File: macros/commands/get_dangling_tables.sql [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+{% macro get_model_schemas_and_tables(deprecated_models_path=none) %}
+  {% set model_schemas = [] %}
+  {% set model_tables = [] %}
+
+  {% set relevant_nodes = [] %}
+  {% for model_node in graph.nodes.values() | selectattr('resource_type', '==', 'model') %}
+    {% if not deprecated_models_path %}
+      {% do relevant_nodes.append(model_node) %}
+    {% elif not model_node.original_file_path.startswith(deprecated_models_path) %}
```
Comment (elongl) [inline macros/commands/get_dangling_tables.sql:45]: Can you just add this as an `or` above?

```j2
    {% if not deprecated_models_path or not model_node.original_file_path.startswith(deprecated_models_path) %}
```

File: macros/commands/get_dangling_tables.sql [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  {% endfor %}
+  {% for db_table in db_tables %}
+    {% if db_table.render() not in rendered_model_tables %}
+      {% do print(db_table) %}
+    {% endif %}
+  {% endfor %}
+{% endmacro %}
+
+
+{% macro get_tables_in_schemas(schemas) %}
```
Comment (elongl) [inline macros/commands/get_dangling_tables.sql:16]: Sorry for the nit-picking, but maybe call this `get_all_tables_in_schemas` in order to clarify that this one includes tables that are not in dbt compared to the macro below?

**Pull request closed without merging**